core/modules/node/node.module

```diff
@@ -2136,6 +2135,33 @@ function node_page_default() {
 }
 
 /**
+ * Helper function to determine callback for /node path.
+ *
+ * @return string
+ *   Returns the string corresponding to the callback function to be used
+ *   for the /node path.
+ */
+function _node_path_or_not() {
+  $front = config_get('system.core', 'site_frontpage');
+  if ($front != 'node') {
+    return 'node_custom_front_page';
+  }
+  else {
+    return 'node_page_default';
+  }
+}
+
+/**
+ * Conditional callback function for $item['node'].
+ * If using a custom front page then redirect the /node path to the custom page.
+ *
+ * @see node_menu()
+ */
+function node_custom_front_page() {
+  backdrop_goto('/', array(), '301');
```

Comment:
This should be `<front>` instead of `/`, which might not work well on Backdrop installs within subdirectories.